### PR TITLE
Rewrite header field name guidelines

### PR DESCRIPTION
**Pull request by mnot**
For #47.

### CODE REVIEW

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -1489,14 +1489,27 @@ deployed and not likely to be registered in a timely manner otherwise.</t>
...
-   (The "X-" prefix idiom has been extensively misused in practice; it was
-   intended to only be used as a mechanism for avoiding name collisions inside
-   proprietary software or intranet processing, since the prefix would ensure
-   that private names never collide with a newly registered Internet name; see
-   <xref target="BCP178"/> for further information).
+   Authors of specifications defining new fields are advised to choose a short
+   but descriptive field name. Short names avoid needless data transmission
+   (especially when a header field is sent often); descriptive names avoid
+   confusion and "squatting" on names that might have broader uses.
+</t>
```
Comment (reschke) [inline draft-ietf-httpbis-semantics-latest.xml]: Isn't that only true for HTTP/1.1?

Comment (mnot) [reply draft-ietf-httpbis-semantics-latest.xml]: No; [HQ]Pack aren't magical, they have limited-size state tables that can be consumed by overly long header names.

Comment (reschke) [reply draft-ietf-httpbis-semantics-latest.xml]: Yes, that's understood. But I think this is still will cause people tp be confused, in particular as it says "when ... sent often".

Comment (mnot) [reply draft-ietf-httpbis-semantics-latest.xml]: OK, removing the parenthetical.

**Pull request merged**